crypto/s2n_ecdsa.c

```diff
@@ -32,8 +35,29 @@
 #include "crypto/s2n_openssl.h"
 #include "crypto/s2n_pkey.h"
 
+
 #define S2N_ECDSA_TYPE 0
 
+S2N_RESULT s2n_unsafe_ecdsa_get_mut(const struct s2n_ecdsa_key *ecdsa_key, EC_KEY **out_ec_key) {
+    RESULT_ENSURE_REF(ecdsa_key);
+    RESULT_ENSURE_REF(ecdsa_key->ec_key);
+    RESULT_ENSURE_EQ(*out_ec_key, NULL);
+
+    /* pragma gcc diagnostic was added in gcc 4.6 */
+#if S2N_GCC_VERSION_AT_LEAST_OR_CLANG(4,6,0)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#endif
+    *out_ec_key = (EC_KEY *) ecdsa_key->ec_key;
+#if S2N_GCC_VERSION_AT_LEAST(4,6,0)
+#pragma GCC diagnostic pop
+#endif
+
+    RESULT_ENSURE_REF(*out_ec_key);
+
```

Comment:
This seems unnecessary. Why shouldn't the key be allowed to be null? Does that affect the conversion, or can it cause a seg fault? Same with the "RESULT_ENSURE_REF(ecdsa_key->ec_key)" above, which is a duplication of this check.